app/src/renderer/indexed_model.rs

```diff
@@ -7,38 +7,67 @@ use wavefront::{Index, Obj, Vertex};
  * Our representation of a model.
  */
 pub struct IndexedModel {
-    pub vertices: Vec<f32>,
-    pub indices: Vec<u16>,
+    pub positions: Vec<f32>,
+    pub position_indices: Vec<u16>,
+    pub tex_coords: Vec<f32>,
+    pub tex_coord_indices: Vec<u16>
 }
 
+pub const BYTE_SIZE_F32: u8 = 4;
+pub const SIZE_POSITION: u8 = 3;
+pub const BYTE_SIZE_POSITION: u8 = BYTE_SIZE_F32 * SIZE_POSITION;
+pub const SIZE_TEX_COORD: u8 = 2;
+pub const BYTE_SIZE_TEX_COORD: u8 = BYTE_SIZE_F32 * SIZE_TEX_COORD;
+
 /**
  * Implementation of the indexed model.
  */
 impl IndexedModel {
     /**
      * Turn a model into its indexed equivalent.
      *
-     * This is done by filling a memotable whose keys are known vertices that we have seen before,
-     * and whose values are indexes into an array containing the position of those vertices.
+     * This is done by filling memotables whose keys are known indexable vertex attributes that we have seen before,
+     * and whose values are indexes into an array containing these vertex attributes.
      */
     pub fn new(obj_data: &Obj) -> IndexedModel {
-        let mut memo: BTreeMap<Index, u16> = BTreeMap::new();
+        let mut vertex_memo: BTreeMap<Index, u16> = BTreeMap::new();
+        let mut tex_coord_memo: BTreeMap<Index, u16> = BTreeMap::new();
         let mut positions = Vec::new();
-        let indices = obj_data
+        let mut tex_coords: Vec<f32> = Vec::new();
+        
+        // Add vertex positions
+        let position_indices = obj_data
             .vertices()
             .map(|vertex| {
-                let vertexId = vertex.position_index();
-                *memo.entry(vertexId).or_insert_with(|| {
-                    let index = (positions.len() / 3) as u16;
+                let vertex_id = vertex.position_index();
+                *vertex_memo.entry(vertex_id).or_insert_with(|| {
+                    let index = (positions.len() / SIZE_POSITION as usize) as u16;
```

Comment:
using as sucks probably better to do 
`u16::try_from(positions.len() / usize::from(SIZE_POSITION)).unwrap()` or similar

Reply:
Thank you! Crashing on overflow instead of implicitly truncating does seem better.

Reply:
Yes, thanks for the suggestion, again :blush: